crates/burn-autodiff/src/runtime/server.rs

```diff
@@ -14,12 +14,26 @@ use alloc::vec::Vec;
 
 #[derive(Default)]
 pub struct AutodiffServer {
-    steps: HashMap<NodeID, StepBoxed>,
-    actions_builder: HashMap<NodeID, CheckpointerBuilder>,
+    steps: HashMap<NodeId, StepBoxed>,
+    actions_builder: HashMap<NodeId, CheckpointerBuilder>,
     memory_management: GraphMemoryManagement,
 }
 
+/// Defines how nodes are clean.
+pub trait NodeCleaner {
+    /// Initialize a new cleaner.
+    fn init() -> Self;
+    /// Cleans a single [node](NodeId).
+    fn clean(&mut self, node: &NodeId);
+}
+
 impl AutodiffServer {
+    pub fn extend(&mut self, other: AutodiffServer) {
+        self.steps.extend(other.steps);
+        self.actions_builder.extend(other.actions_builder);
+        self.memory_management.extend(other.memory_management);
+    }
```

Comment:
A note for the extend is that if they key clashes, the previous value gets removed.